### PR TITLE
Don't mask error in `cquery` transitions output formatter

### PR DESCRIPTION
**Pull request by fmeum**
All errors used to be wrapped in an `InterruptedException` instead of being reported with a message. Also show a warning when an incompatible target is skipped instead of failing the build.

Work towards #21010

### CODE REVIEW

**Comment by fmeum** [comment]
@keith @illicitonion Do you think this handling of incompatible targets is reasonable?

File: src/test/java/com/google/devtools/build/lib/query2/cquery/ProtoOutputFormatterCallbackTest.java [hunk 1]
```diff
@@ -529,6 +529,29 @@ public void outputOnAliasDep() throws Exception {
...
+            genrule(
+                name = "my_target",
+                cmd = "touch $@",
+                outs = ["out"],
+                target_compatible_with = [":incompatible"],
+            )""");
+
+    options.transitions = Transitions.LITE;
+    AnalysisProtosV2.CqueryResult cqueryResult =
+        getProtoOutput("deps(//test:my_target)", AnalysisProtosV2.CqueryResult.parser());
```
Comment (gregestren) [inline src/test/java/com/google/devtools/build/lib/query2/cquery/ProtoOutputFormatterCallbackTest.java:552]: I assume `getProtoOutput` before would trigger the `InterruptedException` that auto-failed this test? Could we, just for clarity, add a line verifying the result isn't null and/or checking the warning message?

Also, what are you distinctly testing here compared to the shell test?

**Comment by keith** [comment]
tested and it seems to work for our case, thanks!

**Review by illicitonion — COMMENTED** [review]
Looks very reasonable - thanks!

File: src/test/java/com/google/devtools/build/lib/query2/cquery/ProtoOutputFormatterCallbackTest.java (hunk 1 above)
Comment (fmeum) [reply src/test/java/com/google/devtools/build/lib/query2/cquery/ProtoOutputFormatterCallbackTest.java]: This was just a leftover from an earlier attempt at a Java test that ultimately failed. I removed it.

**Review by gregestren — APPROVED** [review]

**Comment by fmeum** [comment]
@bazel-io fork 7.4.0

**Comment by iancha1992** [comment]
The changes in this PR have been included in [Bazel 7.4.0 RC1](https://github.com/bazelbuild/bazel/releases/tag/7.4.0rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting `USE_BAZEL_VERSION=7.4.0rc1`. Thanks!

**Pull request closed without merging**